autoload/neomake/makers/ft/clojure.vim

```diff
@@ -0,0 +1,47 @@
+" vim: ts=4 sw=4 et
+
+function! neomake#makers#ft#clojure#EnabledMakers() abort
+    return ['lein']
+endfunction
+
+function! neomake#makers#ft#clojure#lein() abort
+    return {
+                \ 'args': 'check',
+                \ 'append_file': 0,
+                \ 'process_output': function('neomake#makers#ft#clojure#LeinProcessOutput'),
+                \ }
+endfunction
+
+function! neomake#makers#ft#clojure#LeinProcessOutput(entry) abort
+    let errors = []
+    for line in a:entry['output']
+        if line =~# '^Exception'
+            " Example error:
+            " Exception ...: Parameter declaration "!" should be a vector, compiling:(dir/file.cljc:3:1)
+            let error = {'maker_name': 'lein', 'type': 'E', 'bufnr': a:entry.jobinfo.bufnr }
```

Comment:
`maker_name` is not necessary.

And I might change `bufnr` to default to the job's bufnr (but only for file-mode makers / that use `append_file` - which is not the case here):
You might get errors for non-current files here, so you should probably use `filename` instead of `bufnr` here.

Reply:
The default for `type` is `E`, but you can keep it for clarity.